clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -617,7 +618,7 @@ public boolean commitOffsetsSync(Map<TopicPartition, OffsetAndMetadata> offsets,
 
             now = time.milliseconds();
             remainingMs = timeoutMs - (now - startMs);
-        } while (remainingMs > 0);
+        } while (remainingMs > 0 && !isFutureCompleted);
```

Comment:
I don't think this fixes the issue. The issue is the following:

```java
if (!future.isRetriable())
    throw future.exception();
```

It should be:

```java
if (future.failed() && !future.isRetriable())
    throw future.exception();
```

It looks like this may not be the only place where we call `isRetriable` without checking if the future failed. We should consider changing this method to avoid this error altogether. One option would be to replace `succeeded` and `failed` with a status method returning an enum with 3 states: `SUCCEEDED`, `FAILED`, `RETRY`. In most cases, we have to handle these 3 states differently.

@hachikuji, what do you think?